AGENTS.md

```diff
@@ -0,0 +1,495 @@
+```
+
+### Event Hooks System
+
+```python
+from strands.hooks import HookProvider, HookRegistry
+from strands.hooks.events import BeforeInvocationEvent, AfterInvocationEvent
+
+class LoggingHooks(HookProvider):
+    def register_hooks(self, registry: HookRegistry) -> None:
+        registry.add_callback(BeforeInvocationEvent, self.log_start)
+        registry.add_callback(AfterInvocationEvent, self.log_end)
+
+    def log_start(self, event: BeforeInvocationEvent) -> None:
+        print(f"Request started for agent")
+
+    def log_end(self, event: AfterInvocationEvent) -> None:
+        print(f"Request completed for agent")
+
+agent = Agent(hooks=[LoggingHooks()])
+```
+
+### Concurrent Tool Execution
+
+```python
+from strands.tools.executors import ConcurrentToolExecutor
+
+# Execute multiple tools in parallel
+concurrent_executor = ConcurrentToolExecutor()
+agent = Agent(
+    tools=[tool1, tool2, tool3],
+    tool_executor=concurrent_executor
+)
+```
+
+---
+
+## Error Handling and Best Practices
+
+*This section is reserved for future team guidance.*
+
+---
+
+## Testing and Development
+
+### Integration Test Patterns
+
+The codebase includes comprehensive integration tests demonstrating real usage:
+
+- **`test_function_tools.py`**: Basic tool creation and usage
+- **`test_multiagent_swarm.py`**: Multi-agent coordination
+- **`test_mcp_client.py`**: MCP server integration
+- **`test_stream_agent.py`**: Streaming responses
+- **`test_session.py`**: Session persistence
+
+### Development Environment Setup
+
+```bash
+# Create virtual environment
+python -m venv .venv
+source .venv/bin/activate
+
+# Install with development dependencies
+pip install -e ".[dev,all]"
+
+# Run tests
+hatch test                    # Unit tests
+hatch test tests_integ       # Integration tests
+
+# Code quality
+hatch fmt --formatter        # Format code
+hatch fmt --linter          # Lint code
+```
+
+---
+
+## Configuration Reference
+
+### Model Provider Configurations
+
+**AWS Bedrock:**
+```python
+BedrockModel(
+    model_id="us.amazon.nova-pro-v1:0",
+    temperature=0.3,
+    max_tokens=4096,
+    top_p=0.9,
+    streaming=True,
+    guardrails_config={...}
+)
+```
+
+**OpenAI:**
+```python
+from strands.models.openai import OpenAIModel
+
+OpenAIModel(
+    model_id="gpt-4",
+    api_key="sk-...",
+    temperature=0.7,
+    max_tokens=2048,
+    organization="org-..."
+)
+```
+
+**Anthropic:**
+```python
+from strands.models.anthropic import AnthropicModel
+
+AnthropicModel(
+    model_id="claude-3-sonnet-20240229",
+    api_key="sk-ant-...",
+    temperature=0.5,
+    max_tokens=4096
+)
+```
+
+### Tool Executor Options
+
+```python
+# Sequential execution (default)
+from strands.tools.executors import SequentialToolExecutor
+agent = Agent(tool_executor=SequentialToolExecutor())
+
+# Concurrent execution
+from strands.tools.executors import ConcurrentToolExecutor
+agent = Agent(tool_executor=ConcurrentToolExecutor(max_workers=4))
```

Comment:
`max_workers` is not an option.  Perhaps the agent should run mypy on code snippets to ensure they're correct.